plugins/modules/ndo_schema_template_deploy.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Akini Ross (@akinross) <akinross@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: ndo_schema_template_deploy
+short_description: Deploy schema templates to sites
+description:
+- Deploy schema templates to sites.
+author:
+- Akini Ross (@akinross)
+options:
+  schema:
+    description:
+    - The name of the schema.
+    type: str
+    required: yes
+  template:
+    description:
+    - The name of the template.
+    type: str
+    required: yes
+  sites:
+    description:
+    - The name of the site(s).
+    type: list
+    elements: str
+  state:
+    description:
+    - Use C(deploy) to deploy schema template.
+    - Use C(redeploy) to redeploy schema template.
+    - Use C(undeploy) to deploy schema template from a site.
+    - Use C(query) to get deployment status.
+    type: str
+    choices: [ deploy, redeploy, undeploy, query ]
+    default: deploy
+seealso:
+- module: cisco.mso.mso_schema_site
+- module: cisco.mso.mso_schema_template
+extends_documentation_fragment: cisco.mso.modules
+'''
+
+EXAMPLES = r'''
+- name: Deploy a schema template
+  cisco.mso.ndo_schema_template_deploy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    state: deploy
+  delegate_to: localhost
+
+- name: Redeploy a schema template
+  cisco.mso.ndo_schema_template_deploy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    state: redeploy
+  delegate_to: localhost
+
+- name: Undeploy a schema template
+  cisco.mso.ndo_schema_template_deploy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    sites: [ Site1, Site2 ]
+    state: undeploy
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type="str", required=True),
+        template=dict(type="str", required=True),
```

Comment:
You can include schema and template under the required_if.
It may help to query all template objects without specifying the schema and template name.

Reply:
not sure what you mean, but assume you want to see query result of all deploy status? can check it the api allows, else schema and template are required to retrieve the status

Reply:
could not query status generic without schema/template so these are required.